includes/multi-currency/class-multi-currency.php

```diff
@@ -244,10 +292,14 @@ private function initialize_available_currencies() {
 		$woocommerce_currency                                = get_woocommerce_currency();
 		$this->available_currencies[ $woocommerce_currency ] = new Currency( $woocommerce_currency, 1.0 );
 
-		$currencies = $this->get_stored_currencies();
-		foreach ( $currencies as $currency ) {
-			$new_currency                                      = new Currency( $currency[0], $currency[1] );
-			$available_currencies[ $new_currency->get_name() ] = $new_currency;
+		$available_currencies = [];
+
+		$cache_data = $this->get_cached_currencies();
```

Comment:
Sure, would that mean doing something like this in the constructor?

```php
public function __construct( WC_Payments_API_Client $payments_api_client ) {
	$this->payments_api_client = $payments_api_client;
	$this->includes();
	add_action( 'woocommerce_init', [ $this, 'init' ] );
}
```

Reply:
Updated in e3665b5723908ea0060868b4f4575c4167a199fb. I used `init` because there is another hook which is attached at the `init` which requires the Multi Currency initialisation to have already happened. :)

Reply:
One thing to keep in mind here is that the `widgets_init` hook is fired early as part of the `init` hook. So, I think we'll need to register the `init_widgets()` call from the constructor, otherwise, the widget won't be enabled. I'm not sure the same happens with `rest_api_init`, but it's probably worth testing it

Reply:
Thanks @luizreis, great catch, I didn't spot that! Updated in 9a937f58c17f298193737c31eac18ad1547c6b6b to move both into the constructor, I think that it's more suitable to add both actions in `__construct`